h2o-genmodel/src/main/java/hex/genmodel/algos/gam/NBSplinesTypeI.java

```diff
@@ -0,0 +1,176 @@
+package hex.genmodel.algos.gam;
+
+import java.io.Serializable;
+
+import static hex.genmodel.algos.gam.GamUtilsISplines.*;
+
+/**
+ * I implemented the spline described in Section III of doc in JIRA: https://h2oai.atlassian.net/browse/PUBDEV-8398.
+ * Any reference to doc I in the code refer to the one here with the http link.
+ *
+ * The recursive formula in equation 5 is used.  It is implemented as a binary tree with current node with order m
+ * and two child nodes with order m-1.
+ */
+public class NBSplinesTypeI implements Serializable {
+    private static final double EPS = 1e-12;
```

Comment:
Is the `EPS` small enough?  Double can be as small as `4.9e-324`. I know it's unlikely to see data with such a small scale but it could happen, e.g., atomic radius (of smaller atoms) is ~1e-11 m.

I'm not sure how the rest of the calculation would be affected with really small numbers but since we divide them it could be advisable to normalize them. Should we detect things like that and potentially warn the user?

No action is required, I just wanted to bring the potential issue to your attention.

Reply:
I changed the implementation to checking a-b == 0 instead of abs(a-b)<EPS.  This should alleviate your concerns.